cli/js/testing.ts

```diff
@@ -62,70 +62,146 @@ export function test(
   TEST_REGISTRY.push(testDef);
 }
 
-interface TestStats {
-  filtered: number;
-  ignored: number;
-  measured: number;
-  passed: number;
-  failed: number;
-}
-
-export interface RunTestsOptions {
-  exitOnFail?: boolean;
-  failFast?: boolean;
-  only?: string | RegExp;
-  skip?: string | RegExp;
-  disableLog?: boolean;
-  reporter?: TestReporter;
-}
-
-enum TestStatus {
-  Passed = "passed",
-  Failed = "failed",
-  Skipped = "skipped"
-}
-
-interface TestResult {
-  name: string;
-  status: TestStatus;
-  duration: number;
-  error?: Error;
-}
-
-export enum TestEvent {
+enum TestEvent {
```

Comment:
Can this enum be moved into `tests` namespace and exported as well?

Reply:
Are you okay with leaving out the redundant `kind` field from the message interfaces? If so, and we still want to share the `TestEvent` enum between `testing.ts` and userland `test_util.ts`, I insist on putting it behind `Deno.symbols.internal` since it has no meaning in the context of other APIs. If not, the whole PR needs to be scaled back (but I would stress that leaving the `kind` field seems like letting an internal concern bleed into the API).

Reply:
> Are you okay with leaving out the redundant `kind` field from the message interfaces?

Can this issue be addressed in separate PR? I'm not sure it's better that way - `kind` is removed from message interface, but then `TestApi` yields tuples. 

> I insist on putting it behind `Deno.symbols.internal` since it has no meaning in the context of other APIs.

That's right, what's wrong with putting it in `Deno.tests`, it's strictly related to testing?

Reply:
> what's wrong with putting it in `Deno.tests`? It's strictly related to testing

Yeah it should be possible to put it in the namespace while still making sure it can only be referenced by referencing `Deno.symbols.internal`.

> `kind` is removed from message interface, but then `TestApi` yields tuples.

`TestApi` is internal, and honestly `TestEvent` isn't necessary there if that's the issue. Think from the perspective of the public API, you specify a custom reporter to `runTests()`:
```ts
class CustomReporter implements Deno.tests.Reporter {
  async start(message: Deno.tests.StartMessage): Promise<void> {
    // Should message contain a kind field which is constant at "start"?
  }

  async testStart(message: Deno.tests.TestStartMessage): Promise<void> {
    // Should message contain a kind field which is constant at "testStart"?
  }

  async testEnd(message: Deno.tests.TestEndMessage): Promise<void> {
    // Should message contain a kind field which is constant at "testEnd"?
  }

  async end(message: Deno.tests.EndMessage): Promise<void> {
    // Should message contain a kind field which is constant at "end"?
  }
}

Deno.runTests({ reporter: new CustomReporter() });
```

Consider that these message interfaces aren't referenced anywhere else. They're not sent through some common channel where you need to read the `kind` field to tell them apart, they're given to distinct callbacks. They don't have a unifying interface which is used anywhere.

Now say, for my niche userland use case, I wanted `CustomReporter` to forward these messages over a network. Clearly I would need to add some distinguishing field since they're all being sent uniformly through the same data stream. It makes perfect sense for it to be `CustomReporter`'s responsibility to attach a header saying which type of message each one is. This is what `SocketReporter` is. It doesn't make sense for the API to assume you're going to do something like this and prepack each message with a redundant header.

Reply:
Okay, thanks for thorough explanation, it does make sense.

Reply:
I've removed `TestEvent` from `testing.ts`. I'm reasoning that `test_util.ts` and `unit_test_runner.ts` require an enum to ensure consistency across a network, whereas `testing.ts`'s use case is all in the same context so string literal types work perfectly.